File: Sorting/insertion_sort.py
class Solution:
    def insertionSort(self, nums):
        for i in range(1, len(nums)):
            key = nums[i]
            j = i-1
            while(j >= 0 and nums[j] > key):
                nums[j+1] = nums[j]
                j -= 1
            nums[j+1] = key
        
        return nums

if __name__ == "__main__":
    nums = [64, 25, 12, 22, 11]
    print("Original array:", nums)
    Solution().insertionSort(nums)
    print("Sorted array:", nums)

File: Sorting/insertion_sort.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    vector<int> insertionSort(vector<int> &nums)
    {
        for (int i = 1; i < nums.size(); i++)
        {
            int key = nums[i];
            int j = i - 1;
            while (j >= 0 && nums[j] > key)
            {
                nums[j + 1] = nums[j];
                j--;
            }
            nums[j + 1] = key;
        }
        return nums;
    }
};

int main()
{
    vector<int> nums = {5, 2, 9, 1, 5, 6};
    Solution sol;
    vector<int> sorted = sol.insertionSort(nums);

    for (int num : sorted)
        cout << num << " ";
    cout << endl;
    return 0;
}
